starfish/imagestack/imagestack.py

```diff
@@ -383,14 +383,30 @@ def sel(self, indexers: Mapping[Axes, Union[int, tuple]]):
         ImageStack :
             a new image stack indexed by given value or range.
         """
-
-        # convert indexers to Dict[str, (int/slice)] format
-        # TODO shanaxel42 check if this can be changed to xarray.copy(deep=false)
         stack = deepcopy(self)
         selector = indexing_utils.convert_to_selector(indexers)
         stack._data._data = indexing_utils.index_keep_dimensions(self.xarray, selector)
         return stack
 
+    def sel_by_physical_coords(self, indexers: Mapping[Coordinates, Union[float, tuple]]):
```

Comment:
consider replacing `float` with `Number`

also, you can be specific about what the tuple contains, so maybe:

```suggestion
    def sel_by_physical_coords(self, indexers: Mapping[Coordinates, Union[Number, Tuple[Number, Number]]]):
```